### DISCUSSION TITLE
custom listandcount method

### DISCUSSION THREAD

**Discussion by hugo-dem** [medusajs/medusa#4380] [Q&A]
Hi !

I am an intern developer and trying to create a listandcount method to retrieve stores. I have some questions. First, the second param of the listandcount method is supposed to be of type FindConfig<Store>. My store entity is linked to a Page entity. And i would need to retrieve only the page.created_at and not the whole page. But when i do this :
```
const config: FindConfig<Store> = {
    select: ['page.created_at'],
    relations: ['page', 'members'],
    skip: req.query.offset,
    take: req.query.limit
  };
```

it gives me this error: `Type '"page.created_at"' is not assignable to type 'keyof Store'. Did you mean '"created_at"'?`

Isn't it possible to retrieve only the created_at attribute ..?

**Comment by vholik** [comment]
Have you tried this?

```
const config: FindConfig<Store> = {
    select: {
      page: {
       created_at: true
      }
    },
    relations: ['page', 'members'],
    skip: req.query.offset,
    take: req.query.limit
  };
```
